include/chopper/count/check_filenames.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <cassert>
+
+#include <robin_hood.h>
+
+#include <chopper/count/configuration.hpp>
+
+namespace chopper::count
+{
+
+//!\brief Checks the `filename_clusters` for consistent files, wither precomputed or sequence files.
+inline void check_filenames(robin_hood::unordered_map<std::string, std::vector<std::string>> const & filename_clusters,
+                            configuration & config)
+{
+    assert(!filename_clusters.empty());
+    assert(!filename_clusters.begin()->second.empty());
+
+    config.precomputed_files = std::filesystem::path{filename_clusters.begin()->second[0]}.extension() == ".minimizer";
+
+    for (auto const & [key, filenames] : filename_clusters)
+    {
+        for (auto const & filename : filenames)
+        {
+            std::filesystem::path path{filename};
+
+            if (config.precomputed_files && path.extension() != ".minimizer")
+            {
+                throw std::invalid_argument{"You are providing precomputed files but the file " + path.string() +
+                                            " does not have the correct file extension (.minimizer)."
+                                            " Mixing non-/precomputed files is not allowed."};
+            }
+            else if (!config.precomputed_files && path.extension() == ".minimizer")
+            {
+                throw std::invalid_argument{"You are providing sequence files but the file " + path.string() +
+                                            " was identified as a precomputed file (.minimizer)."
+                                            " Mixing non-/precomputed files is not allowed."};
+            }
+        }
```

Comment:
cpp20 would have [ends_with](https://en.cppreference.com/w/cpp/string/basic_string/ends_with) :D

We could use the implementation in some seqan3 validator:

```
    bool case_insensitive_string_ends_with(std::string_view str, std::string_view suffix) const
    {
        size_t const suffix_length{suffix.size()};
        size_t const str_length{str.size()};
        return suffix_length > str_length ?
               false :
               std::ranges::equal(str.substr(str_length - suffix_length), suffix, [] (char const chr1, char const chr2)
               {
                   return std::tolower(chr1) == std::tolower(chr2);
               });
    }
```

Would have the advantage that it's case-insensitive.

The conditions could be expressed by an xor `config.precomputed_files ^ path.extension() == ".minimizer"`. Then the check which exception is thrown could be inside. But not sure if this makes it faster :)

Reply:
I added `case_insensitive_string_ends_with` but I think the performance of checking the filenames is negligible and the two if branches are more expressive than the xor.